baseband_tasks/phases/predictor.py

```diff
@@ -163,18 +158,20 @@ def __call__(self, time, index=None, rphase=None, deriv=0, time_unit=None):
             A phase if ``deriv=0`` and ``rphase=None`` to preserve precision;
             otherwise, a quantity with units of ``cycle / time_unit**deriv``.
         """
-        time_unit = time_unit or u.s
-        if not hasattr(time, 'mjd'):
-            time = Time(time, format='mjd', scale='utc')
+        _user_defined_index = index is not None
+        time = Time(time, format='mjd', scale='utc')
+
         try:  # This also catches index=None
```

Comment:
Looking at this again, I really dislike what I coded here before: if an index is given, it should be used! If you are up for it, could you rewrite as
```
if index is None:
    <check that no time is completely out of range>
    index = self.searchclosest(time)

dt = ...
<no further checks>
```

(If not, I can do it)